TrackingCounter: accept presupplied lock object. Python 3 exec fix.
